packages/wrangler/src/dev/dev.tsx

```diff
@@ -216,13 +216,9 @@ function InteractiveDevSession(props: DevProps) {
 	useTunnel(toggles.tunnel);
 
 	const onReady = (newIp: string, newPort: number, proxyData: ProxyData) => {
-		if (newIp !== props.initialIp || newPort !== props.initialPort) {
-			ip = newIp;
-			port = newPort;
-			if (props.onReady) {
-				props.onReady(newIp, newPort, proxyData);
-			}
-		}
+		ip = newIp;
+		port = newPort;
```

Comment:
I seem to remember needing to add L219 because if I specify port 0, I get back some other port twice or something? idk. Worth checking that the "onReady" only fires once with the final settled port when setting 0.